openlibrary/plugins/worksearch/code.py

```diff
@@ -215,7 +218,7 @@ def run_solr_query(param = {}, rows=100, page=1, sort=None, spellcheck_count=Non
     if q_list:
         if use_dismax:
             q = web.urlquote(' '.join(q_list))
-            solr_select += "&defType=dismax&qf=text+title^5+author_name^5&bf=sqrt(edition_count)^10"
+            solr_select += quote_plus("&defType=dismax&qf=text+title^5+author_name^5&bf=sqrt(edition_count)^10")
```

Comment:
This will encode the `&`s as well; what sort of results do you get/how are you testing this?

Reply:
That's strange that it doesn't work for you, because it worked fine for me. 

You're totally right about the testing though - whoever broke this in the first place should be responsible for writing a unit test that covers it. I just don't know who that was.

Reply:
Or, I'm not asking you to write a test, I'm just curious how you tested it (what search did you run? What page did you visit?). That's what the testing section in the template is useful for :) I haven't run this yet; I just confirmed that `quote_plus('&foo=3')` encodes the `&`, which I imagine will cause problems.

Reply:
```py
>>> quote_plus('&foo=3+4')
'%26foo%3D3%2B4'
```